packages/core/content-manager/admin/src/content-manager.ts

```diff
@@ -209,7 +211,23 @@ class ContentManagerPlugin {
         addDocumentHeaderAction: this.addDocumentHeaderAction.bind(this),
         addEditViewSidePanel: this.addEditViewSidePanel.bind(this),
         getBulkActions: () => this.bulkActions,
-        getDocumentActions: () => this.documentActions,
+        getDocumentActions: (position?: DocumentActionPosition) => {
+          /**
+           * When possible, pre-filter the actions by the components static position property.
+           * This avoids rendering the actions in multiple places where they weren't displayed,
+           * which wasn't visible but created issues with useEffect for instance.
+           * The response should still be filtered by the position, as the static property is new
+           * and not mandatory to avoid a breaking change.
+           */
+          if (position) {
+            return this.documentActions.filter(
+              (action) =>
+                action.position == undefined || [action.position].flat().includes(position)
+            );
+          }
```

Comment:
Can end users register actions? (still unfamiliar with the api and how actions are supposed to be registered :D )

Reply:
Yes they can, that's why I have to make the .position key on the component optional, because we can't make it a requirement now

Reply:
so this is just about fixing the bug when possible, but it will still occur for community registered document actions that don't have it